src/config/migrations.py

```
@@ -0,0 +1,180 @@
+                current,
+            )
+            return data
+        migration = MIGRATIONS[step_from]
+        try:
+            data = migration(data)
+        except Exception:
+            # Re-raise — the caller (ConfigManager.load) decides
+            # whether to log + fall back to defaults. We don't want
+            # to swallow here because silent migration failure would
+            # produce subtly-wrong config at runtime.
+            logger.error(
```

Ruff (0.15.11)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)